src/azure-cli/azure/cli/command_modules/storage/operations/fs_directory.py

```diff
@@ -0,0 +1,38 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""Custom operations for storage file datalake"""
+
+import os
+
+from azure.core.exceptions import HttpResponseError
+from azure.cli.core.profiles import ResourceType
+
+
+def exists(cmd, client):
+    try:
+        client.get_directory_properties()
+        return True
+    except HttpResponseError as ex:
+        from azure.cli.command_modules.storage.sdkutil import _dont_fail_on_exist
+        StorageErrorCode = cmd.get_models("_shared.models#StorageErrorCode",
+                                          resource_type=ResourceType.DATA_STORAGE_FILEDATALAKE)
+        _dont_fail_on_exist(ex, StorageErrorCode.blob_not_found)
```

Comment:
it seems this method should call `_dont_fail_on_error`

Reply:
the method is following the naming in track1 to prevent failing when exists. But name is not a big problem 😊 I am fine to both.